addons/survey/static/src/js/survey_form.js

```diff
@@ -255,6 +262,28 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         }
     },
 
+    /**
+     * Called when an image on an answer in multi-answers question is clicked.
+     * Starts a widget opening a dialog to display the now zoomable image.
+     * this.surveyImgZoom is true if a surveyImageZoomer widget is currently open.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onChoiceImgClick: function (ev) {
+        ev.preventDefault();
+        let imgZoomer = new surveyImageZoomer({
+            sourceImage: $(ev.currentTarget).attr('src')
+        });
+        imgZoomer.on('close_survey_img_zoom', this, this._onCloseSurveyImgZoom);
```

Comment:
Since you properly destroy the associated component when closed, isn't it possible to simply store a reference to the "zoomer" and check that?

So:

```js
this.imgZoomer = new surveyImageZoomer({
    sourceImage: $(ev.currentTarget).attr('src')
});
```

And when checking in the keydown handler:

`if (this.imgZoomer) { ... }`

That way we avoid the custom events?